src/app/components/algorithms/implementation-view/dialogs/implementation-nisq-analyzer-qpu-selection-dialog/implementation-nisq-analyzer-qpu-selection-dialog.component.ts

```diff
@@ -0,0 +1,130 @@
+import { Component, Inject, OnInit } from '@angular/core';
+import {
+  AbstractControl,
+  FormControl,
+  FormGroup,
+  Validators,
+} from '@angular/forms';
+import { EntityModelProviderDto } from 'api-qprov/models/entity-model-provider-dto';
+import {
+  MAT_DIALOG_DATA,
+  MatDialog,
+  MatDialogRef,
+} from '@angular/material/dialog';
+import { ProviderService } from 'api-qprov/services/provider.service';
+
+@Component({
+  selector: 'app-implementation-nisq-analyzer-qpu-selection-dialog',
+  templateUrl:
+    './implementation-nisq-analyzer-qpu-selection-dialog.component.html',
+  styleUrls: [
+    './implementation-nisq-analyzer-qpu-selection-dialog.component.scss',
+  ],
+})
+export class ImplementationNisqAnalyzerQpuSelectionDialogComponent
+  implements OnInit {
+  qpuSelectionFrom: FormGroup;
+  provider?: EntityModelProviderDto;
+  ready?: boolean;
+  isIbmqSelected = true;
+  isSimulatorAllowed = false;
+
+  constructor(
+    public dialogRef: MatDialogRef<
+      ImplementationNisqAnalyzerQpuSelectionDialogComponent
+    >,
+    @Inject(MAT_DIALOG_DATA) public data: DialogData,
+    public dialog: MatDialog,
+    private providerService: ProviderService
+  ) {}
+
+  get vendor(): AbstractControl | null {
+    return this.qpuSelectionFrom.get('vendor');
+  }
+
+  get simulatorAllowed(): AbstractControl | null {
+    return this.qpuSelectionFrom.get('simulatorAllowed');
+  }
+
+  get token(): AbstractControl | null {
+    return this.qpuSelectionFrom.get('token');
+  }
+
+  ngOnInit(): void {
+    this.qpuSelectionFrom = new FormGroup({
+      vendor: new FormControl(this.data.vendor, [
+        // eslint-disable-next-line @typescript-eslint/unbound-method
+        Validators.required,
+      ]),
+      simulatorAllowed: new FormControl(this.data.simulatorAllowed, [
+        // eslint-disable-next-line @typescript-eslint/unbound-method
+        Validators.required,
+      ]),
+      token: new FormControl(this.data.token, [
+        // eslint-disable-next-line @typescript-eslint/unbound-method
+        Validators.required,
+        Validators.maxLength(255),
+      ]),
+    });
+
+    this.vendor.setValue('IBMQ');
```

Comment:
Also a hardcoded TODO

Reply:
It is set to have a default value already set when opening the dialog